app/models/host/discovered.rb

```diff
@@ -146,18 +146,13 @@ def reboot
     proxy_url = self.proxy_url
 
     if proxy_url[:type] == 'proxy'
-      status = ForemanDiscovery::ProxyOperations.new(:url => proxy_url[:url], :operation => 'reboot').parse_put_operation
+      ForemanDiscovery::ProxyOperations.new(:url => proxy_url[:url], :operation => 'reboot').parse_put_operation
     else
-      status = ::ProxyAPI::BMC.new(:url => "http://#{self.ip}:8443").power :action => "cycle"
+      ::ProxyAPI::BMC.new(:url => "http://#{self.ip}:8443").power :action => "cycle"
     end
-
-    msg = status ? 'successful' : 'failed'
```

Comment:
Right, so you're almost back to what was here before, except that you've removed the debug logs.  Not sure why, but I guess it'll work either way.